src/com/amplitude/api/AmplitudeLog.java

```diff
@@ -0,0 +1,106 @@
+package com.amplitude.api;
+
+import android.util.Log;
+
+/**
+ * Created by danieljih on 10/16/15.
+ */
+public class AmplitudeLog {
+
+    private volatile boolean enableLogging = true;
+
+    protected static AmplitudeLog instance = new AmplitudeLog();
+
+    public static AmplitudeLog getLogger() {
+        return instance;
+    }
+
+    private AmplitudeLog() {
+    } // prevent instantiation
+
+    void setEnableLogging(boolean enableLogging) {
+        this.enableLogging = enableLogging;
+    }
+
+    int d(String tag, String msg) {
+        if (enableLogging) return Log.d(tag, msg);
+        return 0;
+    }
+
+    int d(String tag, String msg, Throwable tr) {
+        if (enableLogging) return Log.d(tag, msg, tr);
+        return 0;
+    }
+
+    int e(String tag, String msg) {
+        if (enableLogging) return Log.e(tag, msg);
+        return 0;
+    }
+
+    int e(String tag, String msg, Throwable tr) {
+        if (enableLogging) return Log.e(tag, msg, tr);
+        return 0;
+    }
+
+    String getStackTraceString(Throwable tr) {
+        return Log.getStackTraceString(tr);
+    }
+
+    int i(String tag, String msg) {
+        if (enableLogging) return Log.i(tag, msg);
+        return 0;
+    }
+
+    int i(String tag, String msg, Throwable tr) {
+        if (enableLogging) return Log.i(tag, msg, tr);
+        return 0;
+    }
+
+    boolean isLoggable(String tag, int level) {
+        return Log.isLoggable(tag, level);
+    }
+
+    int println(int priority, String tag, String msg) {
+        return Log.println(priority, tag, msg);
+    }
+
+    int v(String tag, String msg) {
+        if (enableLogging) return Log.v(tag, msg);
+        return 0;
+    }
+
+    int v(String tag, String msg, Throwable tr) {
+        if (enableLogging) return Log.v(tag, msg, tr);
+        return 0;
+    }
+
+    int w(String tag, String msg) {
+        if (enableLogging) return Log.w(tag, msg);
+        return 0;
+    }
+
+    int w(String tag, Throwable tr) {
+        if (enableLogging) return Log.w(tag, tr);
+        return 0;
+    }
+
+    int w(String tag, String msg, Throwable tr) {
+        if (enableLogging) return Log.w(tag, msg, tr);
+        return 0;
+    }
+
+    int wtf(String tag, String msg) {
```

Comment:
lol are you saying wtf to the wtf method. That's a legit method: http://developer.android.com/reference/android/util/Log.html#wtf(java.lang.String, java.lang.Throwable)   it stands for What a Terrible Failure